cfme/utils/appliance/console.py

```diff
@@ -0,0 +1,428 @@
+                        rules_failures.append(rule)
+                    logger.info("{}: {}".format(rule, result[0].text))
+                else:
+                    logger.info("{}: no result".format(rule))
+            else:
+                logger.info("{}: rule not found".format(rule))
+        return rules_failures
+
+    def configure_primary_replication_node(self, pwd):
+        # Configure primary replication node
+        with SSHExpect(self.appliance) as interaction:
+            interaction.send('ap')
+            interaction.answer('Press any key to continue.', '', timeout=20)
+            # 6/8 for Configure Database Replication
+            interaction.answer('Choose the advanced setting: ',
+                               '6' if self.appliance.version < '5.10' else '8')
+            interaction.answer('Choose the database replication operation: ', '1')
+            answer_cluster_related_questions(interaction, node_uid='1',
+                db_name='', db_username='', db_password=pwd)
+            interaction.answer(r'Enter the primary database hostname or IP address: \|.*\| ',
+                               self.appliance.hostname)
+            interaction.answer(resc('Apply this Replication Server Configuration? (Y/N): '), 'y')
+            interaction.answer('Press any key to continue.', '')
+
+    def reconfigure_primary_replication_node(self, pwd):
+        # Configure primary replication node
+        with SSHExpect(self.appliance) as interaction:
+            interaction.send('ap')
+            interaction.answer('Press any key to continue.', '', timeout=20)
+            interaction.answer('Choose the advanced setting: ',
+                               '6' if self.appliance.version < '5.10' else '8')
+            # 6/8 for Configure Database Replication
+
+            interaction.answer('Choose the database replication operation: ', '1')
+            answer_cluster_related_questions(interaction, node_uid='1',
+                db_name='', db_username='', db_password=pwd)
+            interaction.answer(r'Enter the primary database hostname or IP address: \|.*\| ',
+                               self.appliance.hostname)
+            # Warning: File /etc/repmgr.conf exists. Replication is already configured
+            interaction.answer(resc('Continue with configuration? (Y/N): '), 'y')
+            interaction.answer(resc('Apply this Replication Server Configuration? (Y/N): '), 'y')
+            interaction.answer('Press any key to continue.', '')
+
+    def configure_standby_replication_node(self, pwd, primary_ip):
+        # Configure secondary (standby) replication node
+        with SSHExpect(self.appliance) as interaction:
+            interaction.send('ap')
+            interaction.answer('Press any key to continue.', '', timeout=20)
+            interaction.answer('Choose the advanced setting: ',
+                               '6' if self.appliance.version < '5.10' else '8')
```

Comment:
**Optional** since it doesn't change functionality, but we can remove this check.

Reply:
Umn. Quite a good point, but I am also doing upgrade testing and there are some not-automated test doing upgrades on ha setup. There is `test_upgrade_multi_ha_inplace` which needs to be automated and once I do that, it certainly will use this method and the condition will come handy.